packages/core/src/tooltip/Tooltip.tsx

```diff
@@ -99,42 +106,52 @@ export const Tooltip = forwardRef<HTMLDivElement, TooltipProps>(
       reference
     );
 
+    const floatingRef = useForkRef(floating, ref) as Ref<HTMLDivElement>;
+
     return (
       <>
         {isValidElement(children) &&
           cloneElement(children, {
-            ...getTriggerProps(),
+            ...mergeProps(getTriggerProps(), children.props),
             ref: triggerRef,
           })}
 
         {open && !disabled && (
-          <div
-            className={clsx(withBaseName(), withBaseName(status), className)}
-            ref={floating}
-            {...getTooltipProps()}
-          >
-            <div className={withBaseName("container")}>
-              {!hideIcon && (
-                <StatusIndicator
-                  status={status}
-                  size={1}
-                  className={withBaseName("icon")}
-                />
-              )}
-              <span className={withBaseName("content")}>{content}</span>
-            </div>
-            {!hideArrow && (
-              <FloatingArrow
-                {...arrowProps}
-                className={withBaseName("arrow")}
-                strokeWidth={1}
-                fill="var(--salt-container-primary-background)"
-                stroke="var(--tooltip-status-borderColor)"
-                height={5}
-                width={10}
-              />
-            )}
-          </div>
+          <FloatingPortal>
+            <SaltProvider>
```

Comment:
The provider was needed to support the use case where an app has nested modes. The element that is portalled needs to have the same style as the current scope

Reply:
Could these be inline comments? Sounds like good long term 'requirement' type of feature